packages/omgidl-parser/src/IDLNodes/UnionIDLNode.ts

```diff
@@ -0,0 +1,86 @@
+import { IDLNode } from "./IDLNode";
+import { StructMemberIDLNode } from "./StructMemberIDLNode";
+import { AnyIDLNode, IUnionIDLNode } from "./interfaces";
+import { UnionASTNode } from "../astTypes";
+import { INTEGER_TYPES, SIMPLE_TYPES } from "../primitiveTypes";
+import { Case, IDLMessageDefinition, IDLMessageDefinitionField } from "../types";
+
+export class UnionIDLNode extends IDLNode<UnionASTNode> implements IUnionIDLNode {
+  constructor(scopePath: string[], astNode: UnionASTNode, idlMap: Map<string, AnyIDLNode>) {
+    super(scopePath, astNode, idlMap);
+  }
+
+  get type(): string {
+    return this.astNode.name;
+  }
+
+  get isComplex(): boolean {
+    return true;
+  }
+
+  get switchType(): string {
+    let switchType = this.astNode.switchType;
+    if (!SIMPLE_TYPES.has(this.astNode.switchType)) {
+      const typeNode = this.getNode(this.scopePath, this.astNode.switchType);
+      if (typeNode.declarator === "enum" || typeNode.declarator === "typedef") {
+        switchType = typeNode.type;
+      }
+    }
+    if (!isValidSwitchType(switchType)) {
+      throw new Error(`Invalid resolved switch type ${switchType} in ${this.scopedIdentifier}`);
+    }
+    return switchType;
+  }
+
+  get cases(): Case[] {
+    return this.astNode.cases.map((def) => {
```

Comment:
What's the reasoning for implementing these as `get`ters vs. computing them once and storing them as members?